### PR TITLE
add a bool for agent dead

### PR DESCRIPTION
**Pull request by etuleu**
We needed to know whether the agent has died on the agent side.

There are 2 ways the world can end: the agent dies (health goes negative) or the number of steps specified in the egan config was reached. For steps we can infer that on the agent side because we pass the tick counter and the max steps in "step", but we didn't have a way to infer death so we just pass that through.

The reason this works is that egan cycles a few more steps after the agent dies which is actually useful in this case.

### CODE REVIEW

File: models/agent/agent_interface.go [hunk 1]
```diff
@@ -29,5 +29,5 @@ type AgentInterface interface {
 	Init(actionSpace map[string]SpaceSpec, observationSpace map[string]SpaceSpec) (details map[string]string)
 
 	// Step takes in a map of named Observations. It returns a map of named Actions. The observation can be expected to conform to the shape given in Init, and the Action should conform to the action specification given there. The debug string is for debug information from the environment and should not be used for real training or evaluation.
-	Step(observations map[string]etensor.Tensor, debug string) (actions map[string]Action)
+	Step(observations map[string]etensor.Tensor, debug string, isDead bool) (actions map[string]Action)
```
Comment (garymm) [inline models/agent/agent_interface.go:32]: I think this is perhaps not the best name. E.g., if egan ends after a certain number of steps, the agent didn't die, so it's a bit misleading.
How's about something like "newWorld" or "worldStarting" or "agentPresent" (the last one would be a negation)?

**Review by garymm — CHANGES_REQUESTED** [review]

File: models/agent/agent_interface.go (hunk 1 above)
Comment (etuleu) [reply models/agent/agent_interface.go:32]: No, this flag actually means the agent is dead and has been removed from the world. The case where the world just ended but the agent hasn't died can be deduced from the tick counter and max steps which are passed in the observations dict.

**Comment by garymm** [comment]
As discussed, please either close this or change it to `done`

**Comment by etuleu** [comment]
> As discussed, please either close this or change it to `done`

Changed to done. Will update the egan side to make sense for the done sense as well.

File: network/env.thrift [hunk 2]
```diff
@@ -38,5 +38,5 @@ service WorldService {
 service AgentService {
   // The world calls step on the agent, giving it the current state,
   // and expects back an action to perform
-  Actions step(1:Observations observations, 2:string debug)
+  Actions step(1:Observations observations, 2:string debug, 3:bool done)
```
Comment (garymm) [inline network/env.thrift:41]: please add a comment describing `done`

**Review by garymm — APPROVED** [review]

**Pull request merged**